query/Definer.java

```diff
@@ -245,10 +247,18 @@ private void definePlays(ThingType thingType, Set<PlaysConstraint> playsConstrai
         try (ThreadTrace ignored = traceOnThread(TRACE_PREFIX + "defineplays")) {
             playsConstraints.forEach(plays -> {
                 define(plays.relation().get());
-                final RoleType roleType = getRoleType(plays.role().label().get()).asRoleType();
+                LabelConstraint roleTypeLabel = plays.role().label().get();
+                final RoleType roleType = getRoleType(roleTypeLabel).asRoleType();
                 if (plays.overridden().isPresent()) {
-                    final RoleType overriddenType = getRoleType(plays.overridden().get().label().get()).asRoleType();
-                    thingType.setPlays(roleType, overriddenType);
+                    String overriddenLabelName = plays.overridden().get().label().get().properLabel().name();
+                    Optional<? extends RoleType> overriddenType = roleType.getSupertypes()
+                            .filter(rt -> rt.getLabel().name().equals(overriddenLabelName)).findFirst();
+                    if (overriddenType.isPresent()) {
+                        thingType.setPlays(roleType, overriddenType.get());
+
```

Comment:
not a big deal, we can clean this up later, but don't forget to clean up the formatting of the PR, @Phillammon